### PR TITLE
Feat: Video listing

### PR DESCRIPTION
**Pull request by dusariudaykumar**
# Video Listing and Single page video

## Added :
- Single page video
- Listed all videos on the home page

## Previe
videolisting

### CODE REVIEW

File: src/Contexts/video-context.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        const {
+          data: { videos },
+        } = await getVideosService();
+
+        videoDispatch({ type: "INIT_VIDEOS", payload: videos });
+      } catch (error) {
+        console.log(error.message);
+      }
+    })();
+  }, []);
```
Comment (harshitpaliwal95) [inline src/Contexts/video-context.js:25]: good use of IIFE

File: src/Pages/SinglePageVideo/SinglePageVideo.jsx [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            <span>{videoDetails?.published}</span>
+          </div>
+          <div className="icons-wrapper">
+            <BiLike size="2rem" className="like-icon" />
+            <BiDislike size="2rem" className="dislike-icon" />
+          </div>
+        </div>
+      </div>
+
+      <p className="video-description">{videoDetails?.description}</p>
```
Comment (SanketDhabarde) [inline src/Pages/SinglePageVideo/SinglePageVideo.jsx:37]: Suggestion: You can destructure videoDetails

File: src/components/SideNav/SideNav.jsx [hunk 3]
```diff
@@ -12,36 +12,27 @@ const SideNav = () => {
     <aside className="side-nav">
       <ul className="nav-items">
         <li>
-          <NavLink to="/" exact activeClassName="active" className="nav-pill">
+          <NavLink to="/" activeclassname="active" className="nav-pill">
```
Comment (SanketDhabarde) [inline src/components/SideNav/SideNav.jsx:15]: Navlink by default adds an active class to your class list. You can directly access `nav-pill.active`

**Review by SanketDhabarde — APPROVED** [review]
- Ui looks good
- Like the folder structure having services separate

File: src/components/SideNav/SideNav.jsx (hunk 3 above)
Comment (dusariudaykumar) [reply src/components/SideNav/SideNav.jsx:15]: Thanks for the info. I will change this

**Review by harshitpaliwal95 — COMMENTED** [review]
looks good Ui is clean

**Pull request merged**